stablehlo/reference/Element.cpp

```diff
@@ -720,6 +720,57 @@ Element rsqrt(const Element &el) {
       [](std::complex<double> e) { return 1.0 / std::sqrt(e); });
 }
 
+Element sign(const Element &el) {
+  Type type = el.getType();
+
+  if (isSupportedIntegerType(type)) {
+    auto intEl = el.getIntegerValue();
+    if (intEl.isZero()) return Element(type, (int64_t)0);
+    if (intEl.isNegative())
+      return Element(type, (int64_t)-1);
+    else
+      return Element(type, (int64_t)1);
+  }
+
+  if (isSupportedFloatType(type)) {
+    auto elVal = el.getFloatValue();
+
+    if (elVal.isNaN()) return Element(el);
+    if (elVal.isNegZero()) return Element(type, (double)-0.0);
+    if (elVal.isPosZero()) return Element(type, (double)0.0);
+    if (elVal.isNegative()) return Element(type, (double)-1.0);
+    if (!elVal.isNegative()) return Element(type, (double)1.0);
```

Comment:
Drop `if (!elVal.isNegative())` to signify that this is the final case of the chain of conditionals?